tests/torchtune/datasets/test_text_completion_dataset.py

```diff
@@ -45,3 +45,23 @@ def test_get_item(self, mock_load_dataset):
             prompt, label = dataset[i]["tokens"], dataset[i]["labels"]
             assert prompt == self.expected_tokenized_prompts[i]
             assert label == expected_labels[i]
+
+    @mock.patch("torchtune.datasets._text_completion.load_dataset")
+    def test_get_item_no_eos(self, mock_load_dataset):
+        mock_load_dataset.return_value = self.get_samples()
+        expected_labels = self.expected_tokenized_prompts
+
+        dataset = TextCompletionDataset(
```

Comment:
IIUC, this test is just ensuring that a user *can* call the `TextCompletionDataset` with the param `add_eos=False`. However, we would actually want to be able to test that the EOS token is not added in this case.

Reply:
apologies if I'm missing something - does`assert prompt == self.expected_tokenized_prompts[i][:-1]` not test this? I'm trimming the expected output to omit the EOS token

Reply:
Ahhhh, I missed that you were trimming it to omit the EOS token - smart! Can you leave a comment in the test calling that out?

Reply:
for sure! all good now